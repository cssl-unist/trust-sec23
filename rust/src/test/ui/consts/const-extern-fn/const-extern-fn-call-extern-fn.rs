#![feature(const_extern_fn)]

extern "C" {
    fn regular_in_block();
}

const extern fn bar() {
    unsafe {
        regular_in_block();
        //~^ ERROR: calls in constant functions
    }
}

extern fn regular() {}

const extern fn foo() {
    unsafe {
        regular();
        //~^ ERROR: calls in constant functions
    }
}

fn main() {}
