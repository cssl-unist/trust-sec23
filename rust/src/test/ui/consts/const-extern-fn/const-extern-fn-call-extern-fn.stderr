error[E0015]: calls in constant functions are limited to constant functions, tuple structs and tuple variants
  --> $DIR/const-extern-fn-call-extern-fn.rs:9:9
   |
LL |         regular_in_block();
   |         ^^^^^^^^^^^^^^^^^^

error[E0015]: calls in constant functions are limited to constant functions, tuple structs and tuple variants
  --> $DIR/const-extern-fn-call-extern-fn.rs:18:9
   |
LL |         regular();
   |         ^^^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0015`.
